src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4051,6 +4051,17 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
     // Implement visit_Global for Symbol Table visitor.
     void visit_Global(const AST::Global_t &/*x*/) {}
 
+    void visit_AsyncFunctionDef(const AST::AsyncFunctionDef_t &/*x*/){
+        try
+        {
+            // to be implemented
+        }
+        catch(const std::exception& e)
+        {
+            std::cerr << e.what() << '\n';
+        }        
```

Comment:
You have to report the error like this:
```suggestion
            throw SemanticError("The `async` keyword is currently not supported", x.base.base.loc); 
```

Reply:
Thanks @certik. I will make the necessary changes